src/smp.rs

```diff
@@ -41,23 +41,14 @@ static PER_CPU_BASE: Once<SupervisorPageAddr> = Once::new();
 impl PerCpu {
     /// Initializes the `PerCpu` structures for each CPU, taking memory from `mem_map`. This (the
     /// boot CPU's) per-CPU area is initialized and loaded into TP as well.
-    pub fn init(boot_hart_id: u64, mem_map: &mut HwMemMap) {
+    pub fn init(boot_hart_id: u64, hyp_mem: &mut HypPageAlloc) {
```

Comment:
Literally the title of the commit title is: "with the allocator", rather than the hw map. This has been a slowly moving process (that started on per-cpu page table) to move allocation from map to allocator. I think it's clear.